element_calcium_imaging/imaging_preprocess.py

```diff
@@ -0,0 +1,1281 @@
+            params = (ProcessingParamSet * ProcessingTask & key).fetch1("params")
+            segmentation_channel = params.get(
+                "segmentation_channel", caiman_dataset.segmentation_channel
+            )
+
+            fluo_traces = []
+            for mask in caiman_dataset.masks:
+                fluo_traces.append(
+                    {
+                        **key,
+                        "mask": mask["mask_id"],
+                        "fluo_channel": segmentation_channel,
+                        "fluorescence": mask["inferred_trace"],
+                    }
+                )
+
+            self.insert1(key)
+            self.Trace.insert(fluo_traces)
+
+        else:
+            raise NotImplementedError("Unknown/unimplemented method: {}".format(method))
+
+
+@schema
+class ActivityExtractionMethod(dj.Lookup):
+    definition = """
+    extraction_method: varchar(32)
+    """
+
+    contents = zip(["suite2p_deconvolution", "caiman_deconvolution", "caiman_dff"])
+
+
+@schema
+class Activity(dj.Computed):
+    definition = """  # inferred neural activity from fluorescence trace - e.g. dff, spikes
+    -> Fluorescence
+    -> ActivityExtractionMethod
+    """
+
+    class Trace(dj.Part):
+        definition = """  #
+        -> master
+        -> Fluorescence.Trace
+        ---
+        activity_trace: longblob  # 
+        """
+
+    @property
+    def key_source(self):
+        suite2p_key_source = (
+            Fluorescence
+            * ActivityExtractionMethod
+            * ProcessingParamSet.proj("processing_method")
+            & 'processing_method = "suite2p"'
+            & 'extraction_method LIKE "suite2p%"'
+        )
+        caiman_key_source = (
+            Fluorescence
+            * ActivityExtractionMethod
+            * ProcessingParamSet.proj("processing_method")
+            & 'processing_method = "caiman"'
+            & 'extraction_method LIKE "caiman%"'
+        )
+        return suite2p_key_source.proj() + caiman_key_source.proj()
+
+    def make(self, key):
+        method, imaging_dataset = get_loader_result(key, Curation)
+
+        if method == "suite2p":
+            if key["extraction_method"] == "suite2p_deconvolution":
+                suite2p_dataset = imaging_dataset
+                # ---- iterate through all s2p plane outputs ----
+                spikes = []
+                for s2p in suite2p_dataset.planes.values():
+                    mask_count = len(spikes)  # increment mask id from all "plane"
+                    for mask_idx, spks in enumerate(s2p.spks):
+                        spikes.append(
+                            {
+                                **key,
+                                "mask": mask_idx + mask_count,
+                                "fluo_channel": 0,
+                                "activity_trace": spks,
+                            }
+                        )
+
```

Comment:
```suggestion
                spikes = []
                for s2p in suite2p_dataset.planes.values():
                    spikes.extend(
                            dict(
                                key,
                                mask=mask_idx,
                                fluo_channel=0,
                                activity_trace=spks,
                            }
                            for mask_idx, spks in enumerate(s2p.spks, start=len(spikes))
                    )
```

Reply:
Thank you. I will update. There are a couple of errors that I will fix.

- `for spks, spike_idx = enumerate` -> `for spike_idx, spks = enumerate`
- `spike_idx` -> `mask_idx`

Reply:
Updated.